tests/unit/test_wheel.py

```diff
@@ -268,7 +268,27 @@ def test_std_install(self, data, tmpdir):
             scheme=self.scheme,
             req_description=str(self.req),
         )
-        self.assert_installed()
+        self.assert_installed(0o644)
+
+    @pytest.mark.parametrize("user_mask, expected_permission", [
+        (0o27, 0o640)
+    ])
+    def test_std_install_with_custom_umask(self, data, tmpdir,
+                                           user_mask, expected_permission):
+        """Test that the files created after install honor the permissions
+        set when the user sets a custom umask"""
+
+        mask = os.umask(0)
+        os.umask(user_mask)
+        self.prep(data, tmpdir)
+        wheel.install_wheel(
+            self.name,
+            self.wheelpath,
+            scheme=self.scheme,
+            req_description=str(self.req),
+        )
+        self.assert_installed(expected_permission)
+        os.umask(mask)
```

Comment:
I'd suggest this:

```python
prev_umask = os.umask(user_mask)
try:
    ...
finally:
    os.umask(prev_umask)
```

Reply:
That's a nice use of `try.. finally`. Fixed in latest commit